lib/plugins/physics.js

```diff
@@ -309,5 +342,12 @@ function inject (bot, { physicsEnabled }) {
 
   bot.on('mount', () => { shouldUsePhysics = false })
   bot.on('respawn', () => { shouldUsePhysics = false })
+  bot.on('login', () => {
+    shouldUsePhysics = false
+    if (doPhysicsTimer === null) {
+      lastPhysicsFrameTime = performance.now()
+      doPhysicsTimer = setInterval(doPhysics, PHYSICS_SETINTERVAL_MS)
```

Comment:
This feels wrong. When the setInterval schedules the next physics tick to far in the future because of some de synch issue you could end up with the time accumulator having to catch up every time doPhysics is called. This would cause physics to take up way more cpu time then it should. Maybe a solution would be to set a new setTimeout whenever doPhysics ran after the time accumulator had a chance to adjust. And then give the setTimeout an adjusted time when the time accumulator ran late.

Reply:
You basically have two separate clocks running along side each other. But when on clock de syncs and runs late it has to block the event loop to catch up. But there is nothing stopping it from having to do the same thing in the next tick again.

Reply:
This could also explain why sometimes mineflayer is using a lot of cpu time when the bot is just sitting still and doing nothing.

Reply:
Isn't that what doPhysics() is doing?
doPhysics() is a self adjusting loop that is called every 10ms. Every 50ms, it is supposed to call tickPhysics().
It catches up by itself if the setInterval loop falls behind. The problem is that Mineflayer runs the setInterval loop every 50ms, which means doPhysics() isn't doing it's job of playing catchup when the setInterval loop goes over 50ms.

Reply:
> You basically have two separate clocks running along side each other. But when on clock de syncs and runs late it has to block the event loop to catch up. But there is nothing stopping it from having to do the same thing in the next tick again.

This is still a problem with the old design. The old design always had to catchup because setInterval would usually run late.
This is why in the new design, I am trying to incorporate a limiter so that it doesn't run too far behind.

Reply:
But this dose not solve the issue when the two de sync. You should not be using setInterval at all when setInterval can fall behind. Or have a mechanism to adjust either setInterval or the catch up time in doPhysics. 

I agree that this is something that is already present in the current version. But this is a good opportunity to fix it.